includes/utils.h

```diff
@@ -38,7 +38,8 @@ namespace utils {
 		std::ostringstream os;
 		std::copy(begin, end, std::ostream_iterator<std::string>(os, delim.c_str()));
 		std::string ret = os.str();
-		ret.erase(ret.length() - delim.length());
+		if (ret.find(ret.length() - delim.length()) != ret.npos)
+			ret.erase(ret.length() - delim.length());
```

Comment:
きちんと動作確認しましたか？
eraseは引数にret.length()が来ても動作は保証されます。
ret.length()が来た時、何もしません。
よって、以前の状態で正しいです。
むしろ、このfindの使用方法は間違っています。findの引数は文字か文字列であり、数値は入りません。
以前の状態に戻してください。
[eraseのリファレンス](https://cpprefjp.github.io/reference/string/basic_string/erase.html)
[findのリファレンス](https://cpprefjp.github.io/reference/string/basic_string/find.html)

```suggestion
		ret.erase(ret.length() - delim.length());
```

以下(以前の状態での)テストコード
```
	std::vector<std::string> vec;
	vec.push_back("a");
	vec.push_back("b");
	vec.push_back("c");
	vec.push_back("d");
	std::cout << utils::Join(vec.begin(), vec.end(), "") << "|" << std::endl;
```
結果
`abcd|`

Reply:
動作は確認しましたが、動作が不明だったので、もう一度調査

以前の状態でテストをしたところ、TOPIC #ch1 のリプライに下記エラーがでました、
params.size() == 1;
原因はChannel::ExTopicCommand()関数の
this->set_topic(utils::Join(params.begin() + 1, params.end(), " "));ところでした。

utils::Joinで、erase関数にマイナスの値を入れてしまったことが原因でした。
```
ret: ret.size():0
delim:  delim.size():1
ret.erase(ret.length() - delim.length());
```
→再度解消方法を再度見直し、set_topicを行う前にparams.size()>1のことを確認するようにしました。


元々発生したエラー
```
command:TOPIC
command param:
i: 0, size: 4, "#ch1"
vector_length: 2
basic_string::erase: __pos (which is 18446744073709551615) > this->size() (which is 0)
```

Reply:
なるほど、そういうことでしたか。